source/src_lcao/FORCE_k.cpp

```diff
@@ -446,22 +457,49 @@ void Force_LCAO_k::cal_foverlap_k(
 	{
 		edm2d[is] = new double[GlobalC::LNNR.nnr];
 		ModuleBase::GlobalFunc::ZEROS(edm2d[is], GlobalC::LNNR.nnr);
-	}
-	bool with_energy = true;
+    }
+    
+    Record_adj RA;
+	RA.for_2d();
 
 	//--------------------------------------------	
 	// calculate the energy density matrix here.
-	//--------------------------------------------	
-	this->set_EDM_k(edm2d, with_energy);
+    //--------------------------------------------	
+    if (INPUT.new_dm > 0 && INPUT.tddft==0)
+    {
+        ModuleBase::timer::tick("Force_LCAO_k","cal_edm_2d");
```

Comment:
If developer want to trace time for this code block, why not use a new inline function here? Too many lines in one function is not good for reading or testing.